producer.py

```diff
@@ -100,76 +101,119 @@ def _print_version(ctx, _, value) -> None:
     ctx.exit()
 
 
-def _load_package_monitoring_config(config_path: str) -> Optional[Dict[Any, Any]]:
-    """Load package monitoring configuration, retrieve it from a remote HTTP server if needed."""
-    if not config_path:
-        return None
+async def _package_releases_worker(
+    graph: GraphDatabase, package_index: AIOSource, package_name: str
+) -> int:
+    """Async handling of new package releases checks."""
+    try:
+        package_versions = await package_index.get_package_versions(package_name)
```

Comment:
I think you can remove the await here as you use an `async for` loop below which should allow for iterating over the coroutine. See [this](https://stackoverflow.com/questions/56161595/how-to-use-async-for-in-python) stack-overflow. If you do this I think you can remove the need for workers and instead just run the full request. This will also get rid of any need for tuning chunk size.